lib/src/async/retry.dart

```diff
@@ -21,13 +21,22 @@ const _defaultTimeout = const Duration(seconds: 60);
 /// reached.
 ///
 /// Waits [interval] to perform the retry after a failed attempt.
+///
+/// If the task fails to complete once within the timeout it will resolve as an
+/// error with a TimeoutException. If the task completes with an error at least
+/// once before the timeout and never succeeds it will resolve as an error with
+/// the last exception from a task run that did not time out.
```

Comment:
The use case we've had so far has been in integration tests where getting the actual exception at the end is critical. I think it makes sense to have a distinction a timeout (the task took too long the first time) and other failures.